src/App.js

```diff
@@ -2,77 +2,50 @@ import React, { Component } from 'react';
 import './App.css';
 
 class App extends Component {
-  _isMounted = false;
-
   state = {
-    source: 'https://i.imgur.com/VYUdUNJ.png',
+    source: '',
     error: null,
+    loading: true,
   };
 
-  fetchCat() {
-    fetch("https://aws.random.cat/meow").then(res=>res.json()).then(
-      (result) => {
-        this.setState({
-          source: result.file
-        });
-    },
-    (error) => {
-      this.setState({
-        error
-      });
-    }
-    )
-  }
-
   componentDidMount() {
-    this._isMounted = true;
-    const { source } = this.state;
-    this.fetchCat(source);
-
+    this.fetchCat();
   }
 
-  componentWillUnmount() {
-    this._isMounted = false;
+  fetchCat = async () => {
+    try {
+      const result = await fetch("https://aws.random.cat/meow");
+      const image = result.json();
+      this.setState({ source: image.file, loading: false });
+    } catch (e) {
+      this.setState({ error: e.message, loading: false });
+    }
   }
 
   render() {
-    const { error, source } = this.state;
-    if (error) {
-      return <div>Error: {error.message}</div>;
-    }
-    return (
+    const { error, loading, source } = this.state;
+    return loading ? (
+      <p>Loading...</p>
+    ) : error ? (
+      <div>Error: {error}</div>
+    ) : (
```

Comment:
Since we now have loading, i just reworked this into a ternary operator instead of an if statement.

We are checking if loading return <p> loading </p>, otherwise if there is an error, return <p> error </p>, and if there is neither, return our component